m2install.sh

```diff
@@ -215,7 +215,7 @@ function getStripComponentsValue()
 {
     local stripComponents=
     local slashCount=
-    slashCount=$(tar -tf "$1" | grep pub/index.php | grep -v vendor | sed 's/[.][/]pub[/]index[.]php//' | sed 's/pub[/]index[.]php//' | tr -cd '/' | wc -c)
+    slashCount=$(tar -tf "$1" | fgrep pub/index.php | sed 's/pub[/]index[.]php//' | sort | head -1 | tr -cd '/' | wc -c)
```

Comment:
I got different results. Looks like you have missed second sed
[.][/]pub[/]index.php
please consider case, when code archive has paths like
```
tar -tf 2017-05-05_fg6epihtnqmfc.tar.gz | grep pub/index.php
./pub/index.php
./vendor/magento/magento2-base/pub/index.php
```

Probably it should looks like `tar -tf "$1" | fgrep pub/index.php | sed 's/[.][/]pub[/]index[.]php//' | sed 's/pub[/]index[.]php//' | sort | head -1 | tr -cd '/' | wc -c`